airbyte-db/lib/src/main/java/io/airbyte/db/check/DatabaseAvailabilityCheck.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.db.check;
+
+import static org.jooq.impl.DSL.select;
+
+import io.airbyte.commons.lang.Exceptions;
+import io.airbyte.db.Database;
+import java.util.Optional;
+import java.util.function.Function;
+import org.jooq.DSLContext;
+import org.slf4j.Logger;
+
+/**
+ * Performs a check to verify that the configured database is available.
+ */
+public interface DatabaseAvailabilityCheck extends DatabaseCheck {
+
+  /**
+   * The number of times to check if the database is available. TODO replace with a default value in a
+   * value injection annotation
+   */
+  int NUM_POLL_TIMES = 10;
+
+  /**
+   * Checks whether the configured database is available.
+   *
+   * @throws InterruptedException if unable to perform the check.
+   */
+  default void check() throws InterruptedException {
+    var initialized = false;
+    var totalTime = 0;
+    final var sleepTime = getTimeoutMs() / NUM_POLL_TIMES;
+
+    while (!initialized) {
+      getLogger().warn("Waiting for database to become available...");
+      if (totalTime >= getTimeoutMs()) {
```

Comment:
Should we use a library for retrying (like https://resilience4j.readme.io/docs/retry). Instead of a custom retry policy?

Reply:
@benmoriceau Definitely.  However, for this pass, I am trying to simply copy/move the existing logic to the new classes to make it easier to verify like behavior.  Once we get this in place, we can do follow up work to improve this logic.

Reply:
I would most likely push us to use [Failsafe](https://failsafe.dev/) instead of [Resilience4j](https://resilience4j.readme.io/docs/retry), mainly because it is a bit more modern and is supported by the OpenAPI generation, should we decide to turn it on there too.